### DISCUSSION TITLE
istioctl install --set command

### DISCUSSION THREAD

**Discussion by soumyaparui** [istio/istio#45814] [Q&A]
Every time I am trying the following command I am getting getting the error:
**Error:**
`spec.selector: Invalid value: "topology.istio.io\"\\network\"": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character`

**Command:**
```
istioctl install -f  $CFG_FOLDER/istio-eastwestgateway-operator.yaml \
--set revision=$ISTIO_REV \
--set components.ingressGateways[0].name=istio-$EW_GW_NAME \
--set components.ingressGateways[0].label.istio=$EW_GW_NAME \
--set components.ingressGateways[0].label.app=istio-$EW_GW_NAME \
--set "components.ingressGateways[0].label.topology\.istio\.io/network"=$NETWORK \
--set components.ingressGateways[0].k8s.env[name:ISTIO_META_REQUESTED_NETWORK_VIEW].value=$NETWORK \
--set values.global.network=$NETWORK
```

**Comment by howardjohn** [comment]
I'd recommend using `-f` and passing a YAML file. There may be a way to escape it properly but its not worth the pain IMO

**Reply by soumyaparui** [reply]
Thanks for your response.

That I know, but what I was trying to do is parameterizing the installation for different environment while using a generic operator yaml.
